server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -85,19 +107,128 @@ public Builder setStored(boolean value) {
 
         @Override
         protected Parameter<?>[] getParameters() {
-            return new Parameter<?>[] { stored, meta };
+            return new Parameter<?>[] { stored, meta, indexOptions };
         }
 
         @Override
         public SparseVectorFieldMapper build(MapperBuilderContext context) {
+            IndexOptions buildIndexOptions = indexOptions.getValue();
+
+            if (buildIndexOptions == null) {
+                buildIndexOptions = new IndexOptions(
+                    true,
+                    new PruningConfig(PruningConfig.DEFAULT_TOKENS_FREQ_RATIO_THRESHOLD, PruningConfig.DEFAULT_TOKENS_WEIGHT_THRESHOLD)
+                );
+            }
+
             return new SparseVectorFieldMapper(
                 leafName(),
-                new SparseVectorFieldType(context.buildFullName(leafName()), stored.getValue(), meta.getValue()),
-                builderParams(this, context)
+                new SparseVectorFieldType(context.buildFullName(leafName()), stored.getValue(), meta.getValue(), buildIndexOptions),
+                builderParams(this, context),
+                buildIndexOptions
```

Comment:
Curious why `buildIndexOptions` has to be passed in through both the field type and the mapper type here?